proxy/proxy.go

```diff
@@ -467,24 +470,60 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 	}
 }
 
-func selectEndpoint(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
+func fadeIn(now time.Time, duration time.Duration, exponent float64, detected time.Time) float64 {
+	rel := now.Sub(detected)
+	fadingIn := rel > 0 && rel < duration
+	if !fadingIn {
+		return 1
+	}
+
+	return math.Pow(float64(rel)/float64(duration), exponent)
+}
+
+func filterFadeIn(endpoints []routing.LBEndpoint, rt *routing.Route, registry *routing.EndpointRegistry, rnd *rand.Rand) []routing.LBEndpoint {
+	now := time.Now()
+	threshold := rnd.Float64()
+
+	filtered := make([]routing.LBEndpoint, 0, len(endpoints))
+	for _, e := range endpoints {
+		detected := registry.GetMetrics(e.Host).DetectedTime()
+		f := fadeIn(
+			now,
+			rt.LBFadeInDuration,
```

Comment:
There should be a fast path that bails out if there is no fade-in configured for the route.